assume_role.sh

```diff
@@ -1,83 +1,89 @@
 # USAGE:
-# requires 4 args, needs to be run with source to get exported variables to stick
-# source assume_role.sh {sourceAccountNumber} {username} {destinationAccountNumber} {rolename}
+# requires 4 args and optionally a 5th, needs to be run with source to get exported variables to stick
+# source assume_role.sh <sourceAccountNumber> <username> <destinationAccountNumber> <rolename> [durationSeconds]
 
+scriptArgs=$#
 sourceAccountNumber=$1
 username=$2
 destinationAccountNumber=$3
 rolename=$4
+durationSeconds=${5:-3600}
+exitCode=""
```

Comment:
Does this need to be defined up here as well?

Reply:
It technically doesn't, but I thought it was 'better' to declare the global var outside functions, since it ends up being used in the end outside functions? It seemed more clear to me but I can remove it, it's kinda whatevz.